### ISSUE TITLE
twisted version 17 incompatibility?

### ISSUE THREAD

**Issue by nicolaskruchten** [scrapy/scrapy#2648]
I just installed scrapy via conda and `scrapy shell http://nicolas.kruchten.com/` returns the following stack trace:

```
Traceback (most recent call last):
  File "/Users/nicolas/miniconda2/envs/scrapy/bin/scrapy", line 11, in <module>
    sys.exit(execute())
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/scrapy/cmdline.py", line 142, in execute
    _run_print_help(parser, _run_command, cmd, args, opts)
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/scrapy/cmdline.py", line 88, in _run_print_help
    func(*a, **kw)
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/scrapy/cmdline.py", line 149, in _run_command
    cmd.run(args, opts)
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/scrapy/commands/shell.py", line 73, in run
    shell.start(url=url, redirect=not opts.no_redirect)
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/scrapy/shell.py", line 48, in start
    self.fetch(url, spider, redirect=redirect)
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/scrapy/shell.py", line 115, in fetch
    reactor, self._schedule, request, spider)
  File "/Users/nicolas/miniconda2/envs/scrapy/lib/python2.7/site-packages/twisted/internet/threads.py", line 122, in blockingCallFromThread
    result.raiseException()
  File "<string>", line 2, in raiseException
TypeError: 'float' object is not iterable
```

I noticed that conda installs scrapy version 1.1 so I upgraded to scrapy 1.3 which didn't solve the issue.

I then noticed that conda installs twisted version 17.1.0 so I downgraded to twisted 16.6.0 and that resolved the issue. I tried this after having read through the closed issue #2461

To summarize: 
* scrapy 1.1 + twisted 17.1.0 = error
* scrapy 1.3 + twisted 17.1.0 = error
* scrapy 1.3 + twisted 16.6.0 = works

For reference, I'm using Python 2.7.13 on macOS and here is the output of `pip freeze` when I finally got things to work:

```
attrs==15.2.0
Automat==0.5.0
cffi==1.9.1
constantly==15.1.0
cryptography==1.7.1
cssselect==1.0.0
enum34==1.1.6
idna==2.2
incremental==16.10.1
ipaddress==1.0.18
lxml==3.7.3
parsel==1.1.0
pyasn1==0.1.9
pyasn1-modules==0.0.8
pycparser==2.17
PyDispatcher==2.0.5
pyOpenSSL==16.2.0
queuelib==1.4.2
Scrapy==1.3.0
service-identity==16.0.0
six==1.10.0
Twisted==16.6.0
w3lib==1.16.0
zope.interface==4.3.3
```

I would be happy to try various other permutations of versions to try to narrow things down, if that would be helpful :)

**Comment by kmike** [comment]
Latest Scrapy release is 1.3.2, it should work with Twisted 17.

**Comment by nicolaskruchten** [comment]
Ah, great. I didn't realize there were point versions past 1.3 :)

**Comment by 369329303** [comment]
Thank you so much, it works.

**Comment by divick** [comment]
Thanks works for me too now.
